synapse/models/syn.py

```diff
@@ -140,10 +143,21 @@ def _revModl201709191412(self):
             xrefp = '{}:xref'.format(ntyp)
             xrefpint = '{}:xref:intval'.format(ntyp)
             xrefpstr = '{}:xref:strval'.format(ntyp)
+            xrefprop = '{}:xref:prop'.format(ntyp)
             for node in nodes:
                 iden = node[0]
                 srcvtype = node[1].get(xtyp)
-                xrefprop = '{}:xref:prop'.format(ntyp)
+                if srcvtype is None:
+                    # This is expensive node level introspection :(
+                    for prop, valu in s_tufo.props(node).items():
+                        if prop.startswith('xref:'):
+                            form = prop.split('xref:', 1)[1]
+                            if self.core.isTufoForm(form):
+                                srcvtype = form
```

Comment:
should we break after finding a `srcvtype`?

Reply:
Yeah we should

Reply:
fixed